### PR TITLE
TE-137 / 22.12.1 / Adding AD SMB failover test case steps and test and code improvent.

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

File: tests/bdd/ha-bhyve02/test_NAS_T0947.py [hunk 1]
```diff
@@ -35,29 +36,14 @@ def the_browser_is_open_navigate_to_nas_url(driver, nas_url):
...
-    assert wait_on_element(driver, 7, xpaths.sideMenu.dashboard, 'clickable')
-    driver.find_element_by_xpath(xpaths.sideMenu.dashboard).click()
+    rsc.Login_If_Not_On_Dashboard(driver, user, password)
 
 
 @then('You should see the dashboard')
 def you_should_see_the_dashboard(driver):
     """You should see the dashboard."""
     assert wait_on_element(driver, 10, xpaths.dashboard.title)
     assert wait_on_element(driver, 10, xpaths.dashboard.systemInfoCardTitle)
```
Comment (markbainix) [inline tests/bdd/ha-bhyve02/test_NAS_T0947.py:46]: Should systemInfoCardTitle be Snake_Case?

Comment (ericbsd) [reply tests/bdd/ha-bhyve02/test_NAS_T0947.py:46]: I can change that.

File: tests/bdd/ha-bhyve02/test_NAS_T0961.py [hunk 2]
```diff
@@ -163,32 +146,30 @@ def navigate_to_system_setting_and_click_misc(driver):
 def the_miscellaneous_page_should_open(driver):
     """the Advanced page should open."""
     assert wait_on_element(driver, 7, xpaths.advanced.title)
-    assert wait_on_element(driver, 7, xpaths.advanced.systemDatasetPool_card)
+    assert wait_on_element(driver, 7, xpaths.advanced.system_datasetPool_card)
```
Comment (markbainix) [inline tests/bdd/ha-bhyve02/test_NAS_T0961.py:149]: Should system_**datasetPool**_card be Snake_Case ?  ( system_Dataset_Pool_card )

File: tests/bdd/ha-bhyve02/test_NAS_T0961.py [hunk 3]
```diff
@@ -163,32 +146,30 @@ def navigate_to_system_setting_and_click_misc(driver):
...
 @then('click on System Dataset')
 def click_on_system_dataset(driver):
     """click on System Dataset."""
-    assert wait_on_element(driver, 7, xpaths.advanced.systemDatasetPool_configure_button, 'clickable')
-    driver.find_element_by_xpath(xpaths.advanced.systemDatasetPool_configure_button).click()
-    assert wait_on_element(driver, 5, xpaths.popup.warning)
-    assert wait_on_element(driver, 5, xpaths.button.close, 'clickable')
-    driver.find_element_by_xpath(xpaths.button.close).click()
+    assert wait_on_element(driver, 7, xpaths.advanced.system_datasetPool_configure_button, 'clickable')
+    driver.find_element_by_xpath(xpaths.advanced.system_datasetPool_configure_button).click()
```
Comment (markbainix) [inline tests/bdd/ha-bhyve02/test_NAS_T0961.py:156]: Same as above: ( system_Dataset_Pool_Configure_button )

File: tests/bdd/ha-bhyve02/test_NAS_T0961.py [hunk 4]
```diff
@@ -163,32 +146,30 @@ def navigate_to_system_setting_and_click_misc(driver):
...
 def click_on_system_dataser_pool_select_dozer_click_Save(driver, pool_name):
     """click on System Dataset Pool select dozer, click Save."""
-    assert wait_on_element(driver, 5, xpaths.systemDataset.select_pool, 'clickable')
-    driver.find_element_by_xpath(xpaths.systemDataset.select_pool).click()
-    assert wait_on_element(driver, 5, xpaths.systemDataset.pool_option(pool_name))
-    driver.find_element_by_xpath(xpaths.systemDataset.pool_option(pool_name)).click()
+    assert wait_on_element(driver, 5, xpaths.system_dataset.select_pool, 'clickable')
+    driver.find_element_by_xpath(xpaths.system_dataset.select_pool).click()
+    assert wait_on_element(driver, 5, xpaths.system_dataset.pool_option(pool_name))
+    driver.find_element_by_xpath(xpaths.system_dataset.pool_option(pool_name)).click()
```
Comment (markbainix) [inline tests/bdd/ha-bhyve02/test_NAS_T0961.py:172]: Same as above

**Review by markbainix — APPROVED** [review]

**Review by markbainix — APPROVED** [review]

**Pull request merged**